src/main/haskell/kore/src/Kore/Step/Function/Data.hs

```diff
@@ -79,3 +81,21 @@ data AttemptedFunction level variable
 following the same pattern as the other `Common*` types.
 -}
 type CommonAttemptedFunction level = AttemptedFunction level Variable
+
+-- |Yields a pure 'Simplifier' which always returns 'NotApplicable'
+notApplicableFunctionEvaluator
+    :: Simplifier
+         (AttemptedFunction level1 variable, SimplificationProof level2)
+notApplicableFunctionEvaluator = pure (NotApplicable, SimplificationProof)
+
+-- |Yields a pure 'Simplifier' which produces a given 'PureMLPattern'
+purePatternFunctionEvaluator
```

Comment:
Note that this only works because we're not doing proofs right now.

Reply:
Right; when we do proofs, we will presumably need to add a `SimplificationProof` argument to this function, but that seemed like an unnecessary detail at this point.

Reply:
I agree that leaving this as-is is fine, because we're still returning a proof so we'll get a compilation error here when we'll finally do proofs and we'll be forced to fix this, but I just wanted to make sure you and Traian to be aware of this.

Reply:
Doing proofs will probably amount to a complete rewrite of `Step`.